src/schemathesis/runner/impl/core.py

```diff
@@ -197,15 +200,24 @@ def wsgi_test(
 ) -> None:
     # pylint: disable=too-many-arguments
     headers = _prepare_wsgi_headers(headers, auth, auth_type)
-    with catching_logs(LogCaptureHandler(), level=logging.DEBUG) as recorded:
-        start = time.monotonic()
-        response = case.call_wsgi(headers=headers)
-        elapsed = time.monotonic() - start
-    run_targets(targets, elapsed)
-    if store_interactions:
-        result.store_wsgi_response(case, response, headers, elapsed)
-    result.logs.extend(recorded.records)
-    run_checks(case, checks, result, response)
+    for _case in _get_cases(case):
+        with catching_logs(LogCaptureHandler(), level=logging.DEBUG) as recorded:
+            start = time.monotonic()
+            response = _case.call_wsgi(headers=headers)
+            elapsed = time.monotonic() - start
+        run_targets(targets, elapsed)
+        if store_interactions:
+            result.store_wsgi_response(_case, response, headers, elapsed)
+        result.logs.extend(recorded.records)
+        run_checks(_case, checks, result, response)
+
+
+def _get_cases(case: Case) -> Generator[Case, None, None]:
```

Comment:
I think we can do it in a slightly different way and avoid this extra function with `None` as a special value. If we'll factor out a function that should be called on `case` then it will be something like:

```
_network_test(case, ...)  # some args
context = HookContext(case.endpoint)
for case_hook in hooks.get_all_by_name("add_case"):
    _case = case_hook(context, deepcopy(case))
    _network_test(_case, ...)  # some args
```

And we won't need a condition and `None` inside a separate function. Also maybe the hook execution part can be extracted in a separate function as well and be parametrized by `_network_test` / `_wsgi_test` functions that will encapsulate logic from `network_test` / `wsgi_test`